projects/storefrontapp-e2e-cypress/cypress/e2e/regression/asm/asm.emulation.core-e2e.cy.ts

```diff
@@ -78,11 +78,42 @@ context('Assisted Service Module', () => {
       );
     });
 
-    // TODO(#9445): Add e2e test for this scenario
-    it.skip('agent login when user is logged in should start this user emulation', () => {});
+    // TODO(#3974): fix the bug to enable e2e test for this scenario
+    it.skip('agent login when user is logged in should start this user emulation', () => {
+      cy.visit('/login');
+      login(customer.email, customer.password);
+
+      checkout.visitHomePage('asm=true');
+
+      cy.get('cx-asm-main-ui').should('exist');
+      cy.get('cx-asm-main-ui').should('be.visible');
+
+      cy.log('--> Agent logging in');
+      asm.agentLogin('asagent', 'pw4all');
+
+      cy.get('cx-csagent-login-form').should('not.exist');
+      cy.get('cx-customer-selection').should('not.exist');
+      cy.get('cx-customer-emulation').should('be.visible');
+    });
+
+    it('agent logout when user was logged and emulated should restore the session', () => {
+      checkout.visitHomePage('asm=true');
+
+      cy.get('cx-asm-main-ui').should('exist');
+      cy.get('cx-asm-main-ui').should('be.visible');
+
+      cy.log('--> Agent logging in');
+      asm.agentLogin('asagent', 'pw4all');
+
+      cy.log('--> Starting customer emulation');
+      asm.startCustomerEmulation(customer);
 
-    // TODO(#9445): Add e2e test for this scenario
-    it.skip('agent logout when user was logged and emulated should restore the session', () => {});
+      cy.log('--> Agent sign out');
+      asm.agentSignOut();
```

Comment:
Remove the check for `asm.agentSignOut()`, and delete it if it exists elsewhere.

Reply:
This case is to check the behavior of the agent when logging out. If it is removed, this case will lose its meaning

Reply:
Or We can so:
     ```
cy.log('--> Agent sign out');
      asm.agentSignOut();
      cy.get('cx-csagent-login-form').should('exist');
      cy.get('cx-customer-emulation').should('not.exist');
      
      cy.log('--> Restore agent login status to reduce failure rate');
      asm.agentLogin('asagent', 'pw4all');
```
Restore agent login status to reduce failure rate at the end

Reply:
It seems unnecessary to perform a separate check for sign out, as it can potentially cause failures in other ASM E2E tests due to parallel execution.

Reply:
Alternatively, if you feel it is necessary, you can comment it out for now. We have a better solution in place to address the issue of not failing in parallel scenarios. We can revisit this type of testing once we have implemented the improved approach.

Reply:
OK, I have skiped this case, and i created a ticket to track: https://jira.tools.sap/browse/CXMC-7221